Data/HashMap/Base.hs

```diff
@@ -1398,6 +1406,115 @@ alterFEager f !k m = (<$> f mv) $ \fres ->
 {-# INLINABLE alterFEager #-}
 #endif
 
+-- | /O(n*m)/ Inclusion of maps. A map is included in another map if the keys
+-- are subsets and the corresponding values are equal:
+--
+-- >>> isSubmapOf m1 m2 = keys m1 ⊆ keys m2 && and [ v1 == v2 | (k1,v1) <- toList m1; let v2 = m2 ! k1 ]
+--
+-- This defines a partial order on maps. However, 'union' is /not/ its least
+-- upper bound, because it is not cummutative. In particular, for all maps @m1@
+-- and @m2@ it holds @m1 `isSubmapOf` union m1 m2@, however, the other direction
+-- @m2 `isSubmapOf` union m1 m2@ may not hold because 'union' may discard values
+-- of m2.
+--
+-- ==== __Examples__
+--
+-- >>> fromList [(1,'a')] `isSubmapOf` fromList [(1,'a'),(2,'b')]
+-- True
+--
+-- >>> fromList [(1,'a'),(2,'b')] `isSubmapOf` fromList [(1,'a')]
+-- False
+isSubmapOf :: (Eq k, Hashable k, Eq v) => HashMap k v -> HashMap k v -> Bool
+isSubmapOf = isSubmapOfBy (==)
+{-# INLINE isSubmapOf #-}
+
+-- | /O(n*m)/ Inclusion of maps with value comparison. A map is included in
+-- another map if the keys are subsets and the corresponding values are smaller:
+--
+-- >>> isSubmapOfBy (⊑) m1 m2 = keys m1 ⊆ keys m2 && and [ v1 ⊑ v2 | (k1,v1) <- toList m1; let v2 = m2 ! k1 ]
+--
+-- This defines a partial order on maps, for which 'unionWith' is the least
+-- upper bound. More specifically, let @(⊑)@ be a partial order on values for
+-- which @(⊔)@ is the least upper bound. Then for all maps @m1@ and @m2@ it
+-- holds @isSubmapOfBy (⊑) m1 (unionWith (⊔) m1 m2)@ and
+-- @isSubmapOfBy (⊑) m2 (unionWith (⊔) m1 m2)@.
```

Comment:
I'm not a huge fan of using `⊑` as the name of the comparison function.

1. I can't directly type it on my keyboard.
2. It looks very similar to `⊆`

So I'd slightly prefer a name like `cmpV` or similar.

Reply:
I used this symbol, because it is well established in static analyses and order theory (e.g. see https://cs.au.dk/~amoeller/spa/spa.pdf sec 4.2). They use `⊑` as an ordering symbol and `⊔` as least upper bounds symbol for a lattice `(⊑,⊔)`. Furthermore, the similarity of `⊑` to `⊆` is not by accident: Powersets form a lattice with `(⊆,∪)`. Lastly, you don't need to be able to type these symbols on your keyboard, because they only appear in this documentation and not in the code of users.

Reply:
I would prefer to keep this notation for clarity sake.

Reply:
> I used this symbol, because it is well established in static analyses and order theory (e.g. see https://cs.au.dk/~amoeller/spa/spa.pdf sec 4.2). They use `⊑` as an ordering symbol and `⊔` as least upper bounds symbol for a lattice `(⊑,⊔)`. Furthermore, the similarity of `⊑` to `⊆` is not by accident: Powersets form a lattice with `(⊆,∪)`. Lastly, you don't need to be able to type these symbols on your keyboard, because they only appear in this documentation and not in the code of users.

I appreciate the background info! :)

I don't have a strong math background and I don't want to expect users to have it. I think it's important that the documentation is accessible to users from a wide range of backgrounds.

I'm also slightly worried that these mathematical symbols might cause an instinctive reaction like "Oh, it's math – I'm not smart enough for that".

But maybe I'm the outlier here with this view.

@m-renaud, @emilypi What do you think?

Reply:
Ok, I removed them from the documentation.